### PR TITLE
Allwinner D1 recipes refactoring and RTL8723ds fix

### PR DESCRIPTION
**Pull request by landgraf**
Previously there was single allwinner D1 SOC in the layer (Nezha) and recipes were named after it. With adding MangoPi MQ Pro it became a bit messy (especially in configs and dts handling in u-boot and kernel). Renamed u-boot and kernel packages to *-allwinnerd1* and moved board specific parameters/patches into machine conf. 
 
The main goal was to fix Bluetooth device of the MangoPi MQ Pro board (UART connected rtl8723ds). Added firmware recipe and few patches from the LKML submitted few days ago.
**NOTE**: License situation of the firmware is kind of weird and were discussed for example [here]( https://bugs.launchpad.net/ubuntu/+source/rtl8723bt-firmware/+bug/1995462) but Armbian does have firmware in the image (even if bluetooth doesn't work). If it's a problem firmware recipe can be dropped and blobs added manually by the user. 

Tested with MangoPi MQ Pro rev 1.3 (core-image-base, core-image-weston)  - HDMI, Bluetooth, serial console and Wifi work, 
Nezha board should not be affected unless I missed something. Tested only build due to lack of the board.

```
[   32.491153] Bluetooth: hci0: RTL: examining hci_ver=08 hci_rev=000d lmp_ver=08 lmp_subver=8723
[   32.503694] Bluetooth: hci0: RTL: rom_version status=0 version=2
[   32.509974] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723ds_fw.bin
[   32.530475] Bluetooth: hci0: RTL: loading rtl_bt/rtl8723ds_config.bin
[   32.562566] Bluetooth: hci0: RTL: cfg_sz 25, total sz 34605
[   33.133560] Bluetooth: hci0: RTL: fw version 0xaa897e1b

[CHG] Controller 68:B9:D3:6C:90:DB Pairable: yes
```
```
# nmcli dev 
DEVICE         TYPE      STATE         CONNECTION 
wlan0          wifi      connected     SSID_NAME_HERE  
```

### CODE REVIEW

File: recipes-kernel/firmware/linux-firmware-rtl8723ds.bb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+SUMMARY = "WIFI/BT Binary Blobs for rtl8723ds-bt (btrtl module)"
+HOMEPAGE = "https://github.com/Realtek-OpenSource/android_hardware_realtek/tree/rtk1395/bt/rtkbt/Firmware/BT"
+## FIXME
+LICENSE = "\
+    Firmware-rtlwifi_firmware \
+"
```
Comment (alistair23) [inline recipes-kernel/firmware/linux-firmware-rtl8723ds.bb:6]: I feel like we need to have the correct license here

File: recipes-bsp/u-boot/u-boot-allwinnerd1.bb [hunk 2]
```diff
@@ -21,15 +21,14 @@ DEPENDS:append = " \
...
-UBOOT_MACHINE = "nezha_defconfig"
 
 # Overwrite this for your server
 TFTP_SERVER_IP ?= "127.0.0.1"
 
 do_compile[depends] = "opensbi:do_deploy"
 
 do_configure:prepend() {
-    sed -i -e 's,@SERVERIP@,${TFTP_SERVER_IP},g' ${WORKDIR}/tftp-mmc-boot.txt
+    sed -i -e 's,@SERVERIP@,${TFTP_SERVER_IP},g' ${WORKDIR}/tftp-mmc-boot.txt
```
Comment (alistair23) [inline recipes-bsp/u-boot/u-boot-allwinnerd1.bb:31]: Looks like a white space change

File: recipes-kernel/firmware/linux-firmware-rtl8723ds.bb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  install -d -m 0755  ${D}${nonarch_base_libdir}/firmware/rtl_bt/
+  install -Dm 0644 ${S}/rtl8723ds_fw.bin  ${D}${nonarch_base_libdir}/firmware/rtl_bt/
+  install -Dm 0644 ${S}/rtl8723ds_config.bin  ${D}${nonarch_base_libdir}/firmware/rtl_bt/
+}
+
+
+FILES:${PN} =  "${nonarch_base_libdir}/firmware/rtl_bt/"
+INSANE_SKIP =  "arch"
+## Pin to mangopi-mq-pro for now, Other machines must be verified before using
+COMPATIBLE_MACHINE = "mangopi-mq-pro"
```
Comment (alistair23) [inline recipes-kernel/firmware/linux-firmware-rtl8723ds.bb:44]: Missing new line

**Review by alistair23 — COMMENTED** [review]
Overall it looks good!

I think we should sort out the license though. We need to confirm what it is before we can merge it, otherwise it might come back to bite us

File: recipes-kernel/firmware/linux-firmware-rtl8723ds.bb (hunk 3 above)
Comment (landgraf) [reply recipes-kernel/firmware/linux-firmware-rtl8723ds.bb]: Looks like most of the *bb files in the project don't have trailing new line (only ./recipes-bsp/u-boot/u-boot-starfive_v2021.04.bb has). So I rebased this PR to  be consistent (no superfluous new line at the end of the recipe).

**Comment by landgraf** [comment]
>  I think we should sort out the license though. We need to confirm what it is before we can merge it, otherwise it might come back to bite us

Yes, the license situation is unclear. I guess it's because the firmware blob doesn't work with current version of mainline kernel (and should be fixed by 0001-Bluetooth-Add-new-quirk-for-broken-local-ext-feature.patch  0003-drivers-net-bluetooth-Enable-quirk-for-broken-local-.patch 0002-Bluetooth-btrtl-add-support-for-the-RTL8723CS.patch)  so nobody pushed firmware to linux-firmware. Or just nobody cares...
I don't know how to resolve this (contact Realtek?) so I dropped firmware from the PR for now and put a comment into machine conf.  Users (like me :) )  may want to decide to use blobs on their own risk.

**Review by alistair23 — APPROVED** [review]

**Comment by alistair23** [comment]
Thanks for splitting it out. If you want you can open a PR for the firmware, then at least others can find it

**Pull request merged**